packages/ref-metrics-pmp/src/cmip_ref_metrics_pmp/annual_cycle.py

```diff
@@ -0,0 +1,238 @@
+from collections.abc import Iterable
+
+from cmip_ref_core.datasets import FacetFilter, SourceDatasetType
+from cmip_ref_core.metrics import (
+    CommandLineMetric,
+    DataRequirement,
+    MetricExecutionDefinition,
+    MetricExecutionResult,
+)
+from cmip_ref_metrics_pmp.pmp_driver import build_pmp_command, process_json_result
+
+
+class AnnualCycle(CommandLineMetric):
+    """
+    Calculate the annual cycle for a dataset
+    """
+
+    def __init__(self) -> None:
+        self.name = "PMP Annual Cycle"
+        self.slug = "pmp-annual-cycle"
+        self.data_requirements = (
+            DataRequirement(
+                source_type=SourceDatasetType.obs4MIPs,
+                filters=(
+                    FacetFilter(
+                        facets={"source_id": ("GPCP-2-3", "HadISST-1-1"), "variable_id": ("pr", "ts")}
+                    ),
+                ),
+                group_by=("variable_id", "source_id"),
+            ),
+            DataRequirement(
+                source_type=SourceDatasetType.CMIP6,
+                filters=(
+                    FacetFilter(
+                        facets={
+                            "frequency": "mon",
+                            "experiment_id": ("amip", "historical", "hist-GHG", "piControl"),
+                            "variable_id": ("pr", "ts"),
+                        }
+                    ),
+                ),
+                group_by=("variable_id", "source_id", "experiment_id", "variant_label", "member_id"),
+            ),
+        )
+
+        self.parameter_file_1 = "pmp_param_annualcycle_1-clims.py"
+        self.parameter_file_2 = "pmp_param_annualcycle_1-metrics.py"
+
+    def build_cmd(self, params: dict) -> Iterable[str]:
+        """
+        Build the command to run the metric
+
+        Parameters
+        ----------
+        definition
+            Definition of the metric execution
+
+        Returns
+        -------
+            Command arguments to execute in the PMP environment
+        """
+        # Pass the parameters using **kwargs
+        return build_pmp_command(**params)
```

Comment:
I would raise an exception here instead as this method must be defined, but shouldn't be used.

Reply:
I am yet to fully understood this. So this should be defined, but it is okay to not being used. Did I get that right?